shell/platform/embedder/platform_view_embedder.cc

```diff
@@ -4,19 +4,58 @@
 
 #include "flutter/shell/platform/embedder/platform_view_embedder.h"
 
+#include "flutter/fml/make_copyable.h"
+
 namespace flutter {
 
+class PlatformViewEmbedder::EmbedderPlatformMessageHandler
+    : public PlatformMessageHandler {
+ public:
+  EmbedderPlatformMessageHandler(
+      fml::WeakPtr<PlatformView> parent,
+      fml::RefPtr<fml::TaskRunner> platform_task_runner)
+      : parent_(parent), platform_task_runner_(platform_task_runner) {}
+
+  virtual void HandlePlatformMessage(std::unique_ptr<PlatformMessage> message) {
+    platform_task_runner_->PostTask(fml::MakeCopyable(
+        [parent = parent_, message = std::move(message)]() mutable {
+          if (parent) {
+            parent->HandlePlatformMessage(std::move(message));
+          } else {
+            FML_DLOG(WARNING)
+                << "Dropping message on channel " << message->channel();
```

Comment:
This seems like something that falls through the channel buffer logic we have. I'm not quite sure I understand the inter-relationship there. Can you explain it?

Reply:
I'm not suggesting we fully address this in this patch, but maybe we need a bug here, or if not some documentaiton/explanation of why this doesn't apply.

This will end up causing confusing/difficult to debug issues for developers who try to use it and find their messages get dropped (and don't get this log because they're using an opt build).

Reply:
The channel buffers are from Host->Flutter direction.  This relates to the Flutter->Host direction.  The reasons that this else would be executed is that we send a message on the ui thread to an engine that is getting deleted before the message is received on the platform thread.  Dropping them is what we've always done, but we've done it silently in the past.  I don't think we should drop messages silently, so I have a debug print.

Reply:
Ahh, ok. Consider updating this message to say something like `"Message received on channel " << message->channel() << ", but engine is deleted. Dropping message."` or something.

Reply:
I tried to be a bit more succinct: `"Deleted engine dropping message on channel " << message->channel()`